src/native-stack/utils/getDefaultHeaderHeight.tsx

```diff
@@ -7,27 +7,29 @@ const formSheetModalHeight = 56;
 export default function getDefaultHeaderHeight(
   layout: Layout,
   statusBarHeight: number,
-  stackPresentation: StackPresentationTypes
+  stackPresentation: StackPresentationTypes,
+  isLargeHeader = false
 ): number {
   // default header heights
   let headerHeight = Platform.OS === 'android' ? 56 : 64;
 
   if (Platform.OS === 'ios') {
     const isLandscape = layout.width > layout.height;
-    const isFromSheetModal =
+    const isFormSheetModal =
       stackPresentation === 'modal' || stackPresentation === 'formSheet';
-    if (isFromSheetModal && !isLandscape) {
+    if (isFormSheetModal && !isLandscape) {
       // `modal` and `formSheet` presentations do not take whole screen, so should not take the inset.
       statusBarHeight = 0;
     }
 
     if (Platform.isPad || Platform.isTV) {
-      headerHeight = isFromSheetModal ? formSheetModalHeight : 50;
+      headerHeight = isFormSheetModal ? formSheetModalHeight : 50;
     } else {
       if (isLandscape) {
         headerHeight = 32;
       } else {
-        headerHeight = isFromSheetModal ? formSheetModalHeight : 44;
+        if (isFormSheetModal) headerHeight = formSheetModalHeight;
+        headerHeight = isLargeHeader ? 96 : 44;
```

Comment:
The `if` clause has no effect here. Final value of `header` height depends solely on value of `isLargeHeader`. It it should be the case -- please consider removing the `if` clause or adjust the logic.

Reply:
You're right 😅 I forgot about the `else` keyword.

Reply:
Fixed in https://github.com/software-mansion/react-native-screens/pull/1844/commits/ebcbc9574f7c57bf00919940542d48f5fb34869a.